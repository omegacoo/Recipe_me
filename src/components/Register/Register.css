/* Mobile Phone Portrait */
.Register_main {
    box-shadow: 0 -5px 10px -5px;
}

.Register_title {
    margin: 4vh;
    text-align: justify;
    font-size: 5vh;
}

.Register {
    border: 1px solid black;
    box-shadow: 5px 5px 5px 5px;
    width: 75vw;
    text-align: center;
    padding: 5px;
    padding-bottom: 15px;
    margin-left: auto;
    margin-right: auto;
    margin-top: 5vh;
    font-size: 3vh;
    font-weight: bold;
    background-color: #F0EAD6;
}

.Register label {
    font-size: 3vh;
}

.Register input {
    font-size: 3vh;
}

.Register button {
    margin-left: auto;
    margin-right: auto;
    display: block;
    width: 55vw;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    margin-top: 3vh;
    text-decoration: none;
    font-size: 5vh;
    font-weight: bolder;
}

/* iPad 768x1024 Portrait */
@media only screen and (min-width: 768px) and (min-height: 1024px) {
    
}