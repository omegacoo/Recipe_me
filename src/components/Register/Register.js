import React from 'react';

import config from '../../config';

import './Register.css';

export default class Register extends React.Component {
    state = {
        username: '',
        password: '',
        confirmPassword: '',
        email: '',
        error: null
    };

    handleUsernameChange = e => {
        this.setState({
            username: e.target.value
        });
    };

    handlePasswordChange = e => {
        this.setState({
            password: e.target.value
        });
    };

    handleConfirmPasswordChange = e => {
        this.setState({
            confirmPassword: e.target.value
        });
    };
    
    handleEmailChange = e => {
        this.setState({
            email: e.target.value
        });
    };

    handleSubmit = e => {
        e.preventDefault();
        
        if(this.state.password !== this.state.confirmPassword){
            this.setState({
                error: 'passwords must match'
            });
            return false
        };
        
        const myBody = {
            user_name: this.state.username,
            password: this.state.password,
            email: this.state.email
        };

        let myHeaders = new Headers();
        myHeaders.append("Content-Type", "application/json");

        const fetchOptions = {
            method: 'POST',
            body: JSON.stringify(myBody),
            headers: myHeaders
        };

        fetch(config.API_ENDPOINT + '/api/register', fetchOptions)
            .then(res => {
                if(!res.ok){
                    throw new Error(res.statusText)
                };
                console.log('User created successfully');
                this.props.history.push('/');
            })
            .catch(err => {
                console.log(err);
            })
    };

    render(){
        return(
            <div className='Register_main'>
                <h1 className='Register_title'>&nbsp;You are on your way to some great meals! So close now...</h1>
                <form
                    className='Register'
                    onSubmit={this.handleSubmit}
                >
                    <label htmlFor='Register_username'>Username</label>
                    <br/>
                    <input 
                        className='Register_username'
                        type='text'
                        value={this.state.username}
                        onChange={this.handleUsernameChange}
                    />
                    <br/>
                    <label htmlFor='Register_password'>Password</label>
                    <br/>
                    <input 
                        className='Register_password'
                        type='password'
                        value={this.state.password}
                        onChange={this.handlePasswordChange}
                    />
                    <br/>
                    <label htmlFor='Register_confirmPassword'>Confirm Password</label>
                    <br/>
                    <input 
                        className='Register_confirmPassword'
                        type='password'
                        value={this.state.confirmPassword}
                        onChange={this.handleConfirmPasswordChange}
                    />
                    <br/>
                    <label htmlFor='Register_email'>Email</label>
                    <br/>
                    <input 
                        className='Register_email'
                        type='email'
                        value={this.state.email}
                        onChange={this.handleEmailChange}
                    />
                    <br/>
                    <button 
                        type='submit'
                    >
                        Register
                    </button>
                </form>
            </div>
        );
    };
};