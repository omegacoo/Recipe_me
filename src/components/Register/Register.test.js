import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Register from './Register';

describe('Register component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <Register />
            </BrowserRouter>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        expect(
        renderer.create(
            <BrowserRouter>
                <Register />
            </BrowserRouter>).toJSON()
        ).toMatchSnapshot();
    });
});