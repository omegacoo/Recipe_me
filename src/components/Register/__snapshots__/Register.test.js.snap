// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders UI correctly 1`] = `
<div
  className="Register_main"
>
  <h1
    className="Register_title"
  >
    Â You are on your way to some great meals! So close now...
  </h1>
  <form
    className="Register"
    onSubmit={[Function]}
  >
    <label
      htmlFor="Register_username"
    >
      Username
    </label>
    <br />
    <input
      className="Register_username"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <label
      htmlFor="Register_password"
    >
      Password
    </label>
    <br />
    <input
      className="Register_password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <br />
    <label
      htmlFor="Register_confirmPassword"
    >
      Confirm Password
    </label>
    <br />
    <input
      className="Register_confirmPassword"
      onChange={[Function]}
      type="password"
      value=""
    />
    <br />
    <label
      htmlFor="Register_email"
    >
      Email
    </label>
    <br />
    <input
      className="Register_email"
      onChange={[Function]}
      type="email"
      value=""
    />
    <br />
    <button
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
